src/CoreBundle/DependencyInjection/Compiler/AddBackendMainMenuItemProvidersPass.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Chameleon System (https://www.chameleonsystem.com).
+ *
+ * (c) ESONO AG (https://www.esono.de)
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ChameleonSystem\CoreBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class AddBackendMainMenuItemProvidersPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $sidebarMenuItemFactoryDefinition = $container->getDefinition('chameleon_system_core.bridge.chameleon.module.sidebar.menu_item_factory');
+        $menuItemProviderServiceIds = $container->findTaggedServiceIds('chameleon_system.backend_menu_item_provider');
```

Comment:
(Also a possible target for auto-configuration. (?))

Reply:
In libraries/frameworks, autoconfiguration shouldn't be used normally. Also we need to specify an additional attribute, I think this can't be autoconfigured.